hal/armv7m/stm32/l4/timer.c

```diff
@@ -203,13 +231,13 @@ void _hal_timerInit(u32 interval)
 	*(timer_common.lptim + lptim_arr) = 0xffff;
 	hal_cpuDataMemoryBarrier();
 
-	*(timer_common.lptim + lptim_cr) |= 4;
-
 	timer_common.overflowh.f = timer_irqHandler;
 	timer_common.overflowh.n = lptim1_irq;
 	timer_common.overflowh.got = NULL;
 	timer_common.overflowh.data = NULL;
 	hal_interruptsSetHandler(&timer_common.overflowh);
 
+	*(timer_common.lptim + lptim_cr) |= 4;
```

Comment:
I assume that this line enables timer. To keep the order of execution, don't we need a memory barrier?

Reply:
Not strictly needed here, but won't hurt, added